app.js

```diff
@@ -0,0 +1,45 @@
+const express = require("express");
+const { getAllTopics } = require("./controllers/topics.controller");
+const endpoints = require("./endpoints.json");
+const { getArticleById } = require("./controllers/articles.controller");
+
+const app = express();
+
+app.get("/api", (req, res) => {
+  res.json(endpoints);
+});
+
+app.get("/api/topics", getAllTopics);
+
+app.get("/api/articles/:article_id", getArticleById);
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request, invalid id/not a number" });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.msg === "Id not found") {
+    res.status(404).send({ msg: "Id not found" });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
```

Comment:
This is great, do you still need the error handler above on line 24 if you have this one here?